File: ui_examples/views/RecyclerViewExample1/app/src/main/java/cf/domone/android/recyclerview_example_1/MainActivity.java
package cf.domone.android.recyclerview_example_1;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

public class MainActivity extends AppCompatActivity {

    private RecyclerView rvSample;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        this.rvSample = (RecyclerView)findViewById(R.id.rvSample);
        this.rvSample.setLayoutManager(new LinearLayoutManager(this));

        SampleItemAdapter adapter = new SampleItemAdapter(this, SampleItem.getSampleItemList());
        this.rvSample.setAdapter(adapter);
    }
}

File: ui_examples/views/linear_gradient_textview_ex/ex01/src/main/java/lee/dorian/android/linear_gradient_textview_ex01/MainActivity.kt
package lee.dorian.android.linear_gradient_textview_ex01

import android.graphics.Color
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    val views by lazy {
        Views()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        views.tvExample.setTextColorAsLinearGradient(arrayOf(
            Color.parseColor("#FF0000"),
            Color.parseColor("#0000FF")
        ))
    }

    inner class Views(
        val tvExample: TextView = findViewById(R.id.tvExample)
    )
}